host/frontend/host-orchestrator/instancemanager_test.go

```diff
@@ -65,6 +65,79 @@ func TestCreateCVDInvalidRequestsEmptyFields(t *testing.T) {
 	}
 }
 
+func TestCreateCVDNewOperationFails(t *testing.T) {
+	dir := t.TempDir()
+	opName := "operation-1"
+	om := NewMapOM(func() string { return opName })
+	fetchCVDHandler := NewFetchCVDHandler(dir, &FakeFetchCVDDownloader{t: t})
+	im := NewInstanceManager(fetchCVDHandler, om)
+	req := CreateCVDRequest{
+		BuildInfo: &BuildInfo{
+			BuildID: "1234",
+			Target:  "aosp_cf_x86_64_phone-userdebug",
+		},
+		FetchCVDBuildID: "1",
+	}
+
+	im.CreateCVD(req)
+	_, err := im.CreateCVD(req)
+
+	var appErr *AppError
+	if !errors.As(err, &appErr) {
+		t.Errorf("error type <<\"%T\">> not found in error chain", appErr)
+	}
+	var newOperationErr NewOperationError
+	if !errors.As(err, &newOperationErr) {
+		t.Errorf("error type <<\"%T\">> not found in error chain", newOperationErr)
+	}
+	im.launchCVDWG.Wait()
+	expectedOp := OperationData{
+		Name: opName,
+		Done: false,
+	}
+	op, _ := om.GetOperation(opName)
```

Comment:
the idea here was to make that there was an operation created by the first `CreateCVD`call with the relevant name, even though the second call failed. I changed the test a little bit just to reflect that, rather than verifying the complete operation values.

There are independent tests for each of the `MapOM` methods in `operation_test.go` file.